Sketch first part of the lab exercise

NicMcPhee opened the pull request:
This draft write-up gets through what I had envisioned as the "first phase", where we would make a separate contact info page for each person in the lab. I don't expect this will lead to any merge conflicts, but we will have rejected pushes and the need to merge pulls. In my head this was simple and short, but it's gotten kind of long, and I'm not sure if I'm writing too much or too little. Feedback on that would be greatly appreciated.

My idea for the second phase was that we would collectively generate a single contact info document. My assumption is that this will generate merge conflicts, and we'll thus get to deal with them. I'm not entirely sure we'll finish the _first_ phase in the first two hour lab, though, so I'm a little uncertain as to how to proceed.

kklamberty commented on README.md:
```diff
@@ -26,3 +26,134 @@ a single-page app that goes through the major features of `git`
+lab we'll create a new file in that directory for every person in the
+class, containing their personal contact info.
+
+:question: KK – do we want to have person X type up person Y's info
```
I think your idea is a good one. It will also cause people to have to talk and listen... both useful skills for communication.

NicMcPhee replied:
:)

kklamberty commented:
It's a little weird reading markdown this way in order to give feedback on the lab. I imagine that for code it makes a bit more sense, but I wonder if the lack of color coding here will mean that people are looking at code in webstorm rather than here? Or, was it your intent that I do this through webstorm... does it support that?

kklamberty commented on README.md:
```diff
@@ -147,13 +147,97 @@ this is easy:
+work up to the Github repository:
+* `git push` on the command line
+* `VCS -> Git -> Push…` in WebStorm
+This will push all commits on your current branch so a branch with the
```
"to" a branch rather than "so" a branch

kklamberty commented on README.md:
```diff
@@ -1,2 +1,243 @@
+there will be times where one person will need to log out and let their
+partner login so that everyone has a chance to go through all the
+activities.
+
```
Joe used one of those special incognito windows (or similar) when we worked on something for GitHub Classroom, and that also worked just fine. That might be faster.

NicMcPhee replied:
That would work on for the web, but it won't work for editing in WebStorm locally, committing, pushing, etc.

There are ways to hack around that, but they're ugly and error prone, so having people log out/in may be the safest/simplest option.

joethe replied:
I agree. There are ways to commit "as" someone else, but that's a bit of a hassle if you're not used to doing it.

**that said:** This lab doesn't seem to make any mention of actually setting up their local git configurations, which they will likely be prompted to do if they haven't used git before. This is, essentially, just setting your name and email, but it's an important step.

NicMcPhee replied:
Good catch - I tend to forget things like that because I've done them so long ago.

kklamberty commented on README.md:
```diff
@@ -1,2 +1,243 @@
+
+Each contact info page should:
+
+* Be named `<your_name>.md`, e.g., `Nic_McPhee.md`
```
It won't be "your" name, but it will be the partner's name. It's the name of the person whose contact info will be there.

NicMcPhee replied:
Good catch – thanks.

kklamberty commented on README.md:
```diff
@@ -1,2 +1,243 @@
+
+* Stage the new file so `git` knows you want to commit it
+   * `git add <filename>` on the command line
+   * `VCS -> Git -> Add` in WebStorm (although if you used WebStorm to
```
Maybe the part about add in webstorm should be moved or also included above so that people know the answer to that question if it comes up.

NicMcPhee replied:
Good catch. I essentially copied that text up to where they create the file, so they're warned that might happen.

kklamberty commented on README.md:
```diff
@@ -1,2 +1,243 @@
+      your commit message. By default this is `vi/vim` for most people;
+      ask for help if you find yourself trapped in `vim` and can't seem
+      to escape.
+      * You can also use `git commit -m "Your cool commit message"` to
```
Does -m mean you can only write a little? Or, does that just encourage people to only write a little? If we are going to strongly discourage this, I think we should have a page that explains how to change your default editor. I think pico/nano is far simpler and would still allow for some more formatting and such in a comment.

kklamberty replied:
I should clarify that I know that -m is for adding your message - I just don't know if there are restrictions on how long the message can be.

NicMcPhee replied:
I don't think there are restrictions on how long a message can be with `-m`, but as a practical matter people don't write longer (multi-line, or even multi-paragraph) commit messages with `-m`.

We should show people how to set their default editor to pico/nano. It may not matter a lot, though, because most people will probably use WebStorm most of the time, right?

kklamberty commented on README.md:
```diff
@@ -1,2 +1,243 @@
+      avoid being sent to an editor. We _strongly_ discourage this,
+      however, as people rarely enter useful one-line commit messages.
+    * `VCS -> Commit Changes…` in WebStorm. This gives you a nice
+    dialogue where you can select which files you actually
```
"This is useful because you may have made a few changes that are related to different issues - you can make sure your commit makes sense by commenting separately for unrelated changes." Or, something like that. I think this matters enough to write it in the text of the lab.

NicMcPhee replied:
Nice.

kklamberty commented on README.md:
```diff
@@ -1,2 +1,243 @@
+of each contact info page. So in the example above, the first thing
+that Pat would do would be to create a branch for the creation of
+Chris's contact info page.
+
```
This is my test of reviewing

kklamberty commented on README.md:
```diff
@@ -1,2 +1,243 @@
+Once you've created the branch it's time to create your partner's
+contact info page.
+
+Each contact info page should:
```
ssdfasdf

NicMcPhee commented:
> It's a little weird reading markdown this way in order to give feedback on the lab. I imagine that for code it makes a bit more sense, but I wonder if the lack of color coding here will mean that people are looking at code in webstorm rather than here? Or, was it your intent that I do this through webstorm... does it support that?

I don't think that WebStorm provides any support for this kind of review, so it will have to happen on GitHub.

Actually the code coloring *does* show up when doing a review in Github. I think it's not happening here because this is all Markdown, but if you look at [this review](https://github.com/NicMcPhee/a-star-search/pull/18) you can see that the Clojure code is still being colored.

That said, when I did this for class, I did often find myself needing to checkout the code and play with it. There are times (for me) when reading coding isn't enough to convince me that it does (or doesn't) work, and I had to actually run things to see. Well written tests ought to really help with that, though, so maybe this will provide some incentive on that front?

NicMcPhee commented:
@joethe – you have any thoughts on this?

NicMcPhee commented:
@kklamberty – I think this addresses all of your concerns. Note how several of your comments are now "gone" because I changed the code that was referred to in the comment. If you click the little "Show outdated" link it will reveal that bit of code and your comment. I think there are some nice tools for showing what's changed since your last review, so you can more easily focus on just the new changes.
